### PR TITLE
Cleanup unused NetworkLayer errors and logic

### PR DESCRIPTION
**Pull request by Anviking**
# Issue Number

ADP-630

# Overview

- [x] Remove unused `ErrNetworkUnavailable `, `ErrGetBlock`, `ErrGetTxParameters`
    - Does call into question what would happen if the node were to disconnect etc...
- [x] Remove `RetryImmediately` and `RetryLater` from `follow`, which were needed only for jormungandr 

# Comments

### CODE REVIEW

File: lib/core/src/Cardano/Wallet/Network.hs [hunk 1]
```diff
@@ -334,28 +288,24 @@ follow nl tr cps yield header =
...
 
-    step :: Int -> Bool -> Cursor -> IO FollowExit
-    step delay hasRolledForward cursor = runExceptT (nextBlocks nl cursor) >>= \case
-        Left e -> do
-            traceWith tr $ MsgNextBlockFailed e
-            sleep (retryDelay delay) hasRolledForward cursor
-
-        Right AwaitReply -> do
+    step :: Bool -> Cursor -> IO FollowExit
+    step hasRolledForward cursor = nextBlocks nl cursor >>= \case
```
Comment (Anviking) [inline lib/core/src/Cardano/Wallet/Network.hs:292]: delay argument not needed anymore

**Review by rvl — APPROVED** [review]
Great, I like clean ups.

`FollowAction(ExitWith)` and `FollowExit(FollowFailure, FollowDone)` could also be removed, but that's a PR for another day...

**Comment by rvl** [comment]
bors r+

Please

**Pull request merged**